src/api/commands/itemimage/ItemImageFeaturedCommand.ts

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2017-2019, The Particl Market developers
+// Distributed under the GPL software license, see the accompanying
+// file COPYING or https://github.com/particl/particl-market/blob/develop/LICENSE
+
+import { inject, named } from 'inversify';
+import { validate, request } from '../../../core/api/Validate';
+import { Logger as LoggerType } from '../../../core/Logger';
+import { Types, Core, Targets } from '../../../constants';
+import { ItemImageService } from '../../services/ItemImageService';
+import { RpcRequest } from '../../requests/RpcRequest';
+import { RpcCommandInterface } from '../RpcCommandInterface';
+import { MessageException } from '../../exceptions/MessageException';
+import { Commands} from '../CommandEnumType';
+import { BaseCommand } from '../BaseCommand';
+import { ListingItemTemplateService } from '../../services/ListingItemTemplateService';
+
+export class ItemImageFeaturedCommand extends BaseCommand implements RpcCommandInterface<void> {
+
+    public log: LoggerType;
+
+    constructor(
+        @inject(Types.Core) @named(Core.Logger) public Logger: typeof LoggerType,
+        @inject(Types.Service) @named(Targets.Service.ItemImageService) private itemImageService: ItemImageService,
+        @inject(Types.Service) @named(Targets.Service.ListingItemTemplateService) private listingItemTemplateService: ListingItemTemplateService
+    ) {
+        super(Commands.ITEMIMAGE_FEATURED);
+        this.log = new Logger(__filename);
+    }
+
+    /**
+     * data.params[]:
+     *  [0]: featured_item_image_ID
+     *  [1]: image_ID
+     * @param data
+     * @returns {Promise<ItemImage>}
+     */
+    @validate()
+    public async execute( @request(RpcRequest) data: RpcRequest): Promise<void> {
+        if (data.params.length < 2) {
```

Comment:
add validate() function and validate the params there. check if all required params exist and that they are valid. use MissingParamException and InvalidParamException instead of MessageException.

Reply:
like this:

        if (data.params.length < 1) {
            throw new MissingParamException('listingItemHash');
        } else if (data.params.length < 2) {
            throw new MissingParamException('profileId');
        }
        if (typeof data.params[0] !== 'string') {
            throw new InvalidParamException('listingItemHash', 'string');
        } else if (data.params[1] !== 'number') {
            throw new InvalidParamException('profileId', 'number');
        }

then in blackbox tests, validate the error message content using new Exception().getMessage();

Reply:
you could also add integration test case for the service.setFeaturedImage